### PR TITLE
Feat/drawer navigator

### PR DESCRIPTION
**Pull request by Leeosteves**
# 🚨 Fill in the fields below

[Add Drawer Navigation](https://trello.com/c/hkFabkmN/43-add-drawer-navigation)

## Description

```
~>Added navigation drawer with functions to switch screens.
~>Added avatar component
~>Added logout button in drawer along with favorites screen
```

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

- [x] PRODUCT

  - [x] Put PR link on ticket
  - [x] Move the ticket

- [x] DEVELOPERS

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings and I added `TODO:`, `FIXME:`

- [x] REVIEW
  - [x] Add reviewers: The entire team and most influential people in the area/project
  - [x] Assignees myself to it

🚨 - `Approve: Wait at least one influential in the project or more experienced`

# FRONTEND

## Screenshot or video of your application

🚨 - `Fill with your application`

https://user-images.githubusercontent.com/106574167/180667562-0fd9f73e-547b-43a3-ac62-e8ea7831310d.mp4

### CODE REVIEW

File: src/screens/Drawer/DrawerContent.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        <Text size={16} isBold>
+          Welcome
+        </Text>
+        <Text size={16} isBold>
+          {googleUser?.familyName}
+        </Text>
+        {/* <TouchableOpacity
+          onPress={() => Linking.openURL('https://hotmail.com')}>
+          <Text size={10}>{googleUser?.email}</Text>
+        </TouchableOpacity> */}
```
Comment (IbraDajani) [inline src/screens/Drawer/DrawerContent.tsx:42]:
```suggestion
      
```

Remove comments

File: src/screens/Drawer/DrawerContent.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      </ProfileContainer>
+      <ButtonContainer>
+        <RouteNavigationButton
+          onPress={() => navigate('Home')}
```
Comment (IbraDajani) [inline src/screens/Drawer/DrawerContent.tsx:46]: Create a functions to deal with this navigation and then pass it on this onPress

File: src/screens/Drawer/DrawerContent.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        />
+        <RouteNavigationButton
+          onPress={() => navigate('Favorites')}
+          icon="favorite"
+          routeName="Favorites"
+          setCurrentTab={setCurrentTab}
+          currentTab={currentTab}
+        />
+        <RouteNavigationButton
+          onPress={() => {}}
```
Comment (IbraDajani) [inline src/screens/Drawer/DrawerContent.tsx:60]: Remove anonymous functions

File: src/screens/Drawer/DrawerContent.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          icon="darkmode"
```
Comment (IbraDajani) [inline src/screens/Drawer/DrawerContent.tsx:61]: remove ir

File: src/screens/Drawer/localComponents/RouteNavigationButton/types.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import {TouchableOpacityProps} from 'react-native';
+import {NameIconTypes} from '~/constants/icons';
+
+export type RouteName = 'Home' | 'Favorites' | 'Dark Mode';
```
Comment (IbraDajani) [inline src/screens/Drawer/localComponents/RouteNavigationButton/types.ts:5]:
```suggestion
export type RouteName = 'Home' | 'Favorites' ;
```

Comment (IbraDajani) [reply src/screens/Drawer/localComponents/RouteNavigationButton/types.ts:5]: Remove the Dark type

File: src/screens/Drawer/styles.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import styled from 'styled-components/native';
+import Avatar from '~/components/Avatar';
+import Text from '~/components/Text';
+
+export const Container = styled.View`
+  flex: 1;
+  flex-direction: column;
```
Comment (IbraDajani) [inline src/screens/Drawer/styles.ts:7]: the default flex direction on React native is column... there is no need to add it on the View

File: src/screens/Drawer/styles.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+`;
+export const ProfileContainer = styled.View`
+  align-items: center;
+`;
+export const AvatarProfile = styled(Avatar)``;
+export const ButtonContainer = styled.View`
+  flex-direction: column;
+  margin-bottom: 170px;
+  /* background-color: gray; */
+  /* border-radius: 24px; */
```
Comment (IbraDajani) [inline src/screens/Drawer/styles.ts:22]: Remove comments

File: src/screens/Drawer/styles.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  top: -75px;
+  position: absolute;
+  opacity: 0.1;
+`;
+export const ButtonLogout = styled.TouchableOpacity`
+  min-width: 51%;
+  flex-direction: row;
+  align-items: center;
+  padding: 8px;
+  /* padding-left: 15px; */
```
Comment (IbraDajani) [inline src/screens/Drawer/styles.ts:37]: Remove comments

File: src/screens/Favorites/index.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Header from '~/components/Header';
+import {BackgroundImage, Container} from './styles';
+import ButtonHeader from '~/components/ButtonHeader';
+import Separator from '~/components/Separator';
+import {useNavigation} from '@react-navigation/native';
+import pokeBallImage from '~/assets/images/pokeball.png';
```
Comment (IbraDajani) [inline src/screens/Favorites/index.tsx:7]: Sort import order by its type

File: src/screens/Favorites/index.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+const Favorites = () => {
+  const {goBack} = useNavigation();
+  const navigation = useNavigation();
+  return (
+    <Container>
+      <BackgroundImage source={pokeBallImage} />
+      <Separator height={54} />
+      <Header>
+        <ButtonHeader icon="menu" onPress={() => navigation.toggleDrawer()} />
```
Comment (IbraDajani) [inline src/screens/Favorites/index.tsx:17]: Create a function to handle this toggle and pass it on this onPress

File: src/screens/Favorites/index.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        <ButtonHeader icon="closedEye" onPress={() => goBack()} />
```
Comment (IbraDajani) [inline src/screens/Favorites/index.tsx:18]: here too

File: src/screens/Favorites/styles.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import {Image} from 'react-native';
+import styled from 'styled-components/native';
+
+export const Container = styled.View`
+  flex: 1;
+  align-items: center;
+  padding: 0 10px;
+`;
+export const BackgroundImage = styled(Image)`
```
Comment (IbraDajani) [inline src/screens/Favorites/styles.ts:9]:
```suggestion
export const BackgroundImage = styled.Image`
```

THERE IS NO NEED TO IMPORT THE COMPONENT FROM THE REACT NATIVE IF YOU CAN JUST USE IT LIKE STYLED.COMPONENT

File: src/screens/Favorites/styles.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import {Image} from 'react-native';
```
Comment (IbraDajani) [inline src/screens/Favorites/styles.ts:1]: remove it

File: src/screens/Home/styles.ts [hunk 14]
```diff
@@ -1,6 +1,7 @@
 import styled from 'styled-components/native';
 import Text from '~/components/Text';
 import {RFValue} from 'react-native-responsive-fontsize';
+import {Image} from 'react-native';
```
Comment (IbraDajani) [inline src/screens/Home/styles.ts:4]: REMOVE IT

File: src/screens/Home/styles.ts [hunk 15]
```diff
@@ -19,7 +20,7 @@ export const FlatListContainer = styled.View`
   padding-bottom: 200px;
 `;
 
-export const BackgroundImage = styled.Image`
+export const BackgroundImage = styled(Image)`
```
Comment (IbraDajani) [inline src/screens/Home/styles.ts:23]:
```suggestion
export const BackgroundImage = styled.Image`
```

**Review by IbraDajani — CHANGES_REQUESTED** [review]

**Comment by IbraDajani** [comment]
REMOVE THE POKEBALL IMAGE FROM THE DRAWER NAVIGATOR... JUST LEAVE THE POKEBALL IMAGE FROM THE HOMESCREEN

**Comment by IbraDajani** [comment]
And fix the drawer navigation type

**Comment by IbraDajani** [comment]
And add the params type too

**Comment by IbraDajani** [comment]
Create a ViewModel to every screen created in these commits

File: src/constants/icons.ts [hunk 16]
```diff
@@ -5,6 +5,10 @@ import closedEye from '~/assets/icons/closedEye/Vector.png';
 import openedEye from '~/assets/icons/openedEye/Vector.png';
 import logout from '~/assets/icons/logout/Vector.png';
 import google from '../assets/icons/google/Vector.png';
+import home from '../assets/icons/home/Vector.png';
+import settings from '../assets/icons/settings/Vector.png';
+import darkmode from '../assets/icons/darkmode/Vector.png';
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/constants/icons.ts]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/constants/icons.ts [hunk 17]
```diff
@@ -13,6 +17,21 @@ export type NameIconTypes =
   | 'closedEye'
   | 'openedEye'
   | 'logout'
-  | 'google';
+  | 'google'
+  | 'home'
+  | 'darkmode'
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/constants/icons.ts]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/constants/icons.ts [hunk 18]
```diff
@@ -13,6 +17,21 @@ export type NameIconTypes =
...
+  filter,
+  like,
+  menu,
+  closedEye,
+  openedEye,
+  google,
+  logout,
+  home,
+  settings,
+  darkmode,
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/constants/icons.ts]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/navigation/DrawerNavigationRoutes/index.tsx [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+      drawerContent={() => <DrawerContent />}
+      screenOptions={{
+        headerShown: false,
+        drawerPosition: 'right',
+        drawerType: 'slide',
+        overlayColor: 'transparent',
+        drawerStyle: {
+          width: 205,
+        },
+      }}>
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/navigation/DrawerNavigationRoutes/index.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/navigation/StackNavigatorRoutes/SignedInRoutes/index.tsx [hunk 20]
```diff
@@ -1,14 +1,14 @@
 import React from 'react';
+import Internal from '../../../screens/Internal';
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/navigation/StackNavigatorRoutes/SignedInRoutes/index.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx [hunk 21]
```diff
@@ -0,0 +1,80 @@
+import React, {useState} from 'react';
+import {ActivityIndicator, View} from 'react-native';
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+  ProfileContainer,
+} from './styles';
+import pokeBallImage from '~/assets/images/pokeball.png';
+import useAuth from '~/hooks/useAuth';
+import Separator from '~/components/Separator';
+import Icon from '~/components/Icon';
+import {useNavigation} from '@react-navigation/native';
+import Text from '~/components/Text';
+import {RouteName} from './localComponents/RouteNavigationButton/types';
+import RouteNavigationButton from './localComponents/RouteNavigationButton';
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx (hunk 1 above)
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx (hunk 2 above)
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+
+const DrawerContent = () => {
+  const {googleUser, handleGoogleSignOut, loading} = useAuth();
+  const {navigate} = useNavigation();
+  const [currentTab, setCurrentTab] = useState<RouteName>('Favorites');
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx (hunk 3 above)
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Drawer/DrawerContent.tsx (hunk 4 above)
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 30a06e50e485c45aaf6a9e54390391fee0ddfd33

File: src/screens/Favorites/useFavoritesController.ts [hunk 24]
```diff
@@ -0,0 +1,11 @@
+import {DrawerActions, useNavigation} from '@react-navigation/native';
+
+const useFavoritesController = () => {
+  const navigation = useNavigation();
+  const openDrawer = () => {
+    navigation.dispatch(DrawerActions.toggleDrawer());
```
Comment (IbraDajani) [inline src/screens/Favorites/useFavoritesController.ts:6]:
```suggestion
  dispatch(DrawerActions.toggleDrawer());
```

File: src/screens/Drawer/styles.ts [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+`;
+export const ButtonLogoutText = styled(Text).attrs({
+  size: 14,
+  isBold: true,
+})``;
+export const Loading = styled(Animated.Image)`
+  width: 25px;
+  height: 25px;
+  align-items: center;
+  /* position: absolute;
```
Comment (IbraDajani) [inline src/screens/Drawer/styles.ts:45]: Remove comments

File: src/screens/Drawer/styles.ts [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+  right: 20px; */
```
Comment (IbraDajani) [inline src/screens/Drawer/styles.ts:46]: Remove comments

File: src/screens/Favorites/index.tsx [hunk 27]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Header from '~/components/Header';
+import ButtonHeader from '~/components/ButtonHeader';
+import Separator from '~/components/Separator';
+import pokeBallImage from '~/assets/images/pokeball.png';
+import {BackgroundImage, Container} from './styles';
+import useFavoritesController from './useFavoritesController';
+
+const Favorites = () => {
+  const {openDrawer} = useFavoritesController();
```
Comment (IbraDajani) [inline src/screens/Favorites/index.tsx:10]: /**
* Hooks
/**

File: src/screens/Favorites/styles.ts [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+
+export const Container = styled.View`
+  flex: 1;
+  align-items: center;
+  padding: 0 10px;
+`;
+export const BackgroundImage = styled.Image`
+  width: 285px;
+  height: 285px;
+  left: 235px;
```
Comment (IbraDajani) [inline src/screens/Favorites/styles.ts:11]: Change this to a right property with a more precise value

File: src/screens/Home/useHomeController.ts [hunk 29]
```diff
@@ -0,0 +1,11 @@
+import {useNavigation, DrawerActions} from '@react-navigation/native';
+
+const useHomeController = () => {
+  const navigation = useNavigation();
```
Comment (IbraDajani) [inline src/screens/Home/useHomeController.ts:4]:
```suggestion
  const {dispatch} = useNavigation();
```

File: src/screens/Home/useHomeController.ts [hunk 30]
```diff
@@ -0,0 +1,11 @@
...
+  const openDrawer = () => {
+    navigation.dispatch(DrawerActions.toggleDrawer());
```
Comment (IbraDajani) [inline src/screens/Home/useHomeController.ts:6]:
```suggestion
    dispatch(DrawerActions.toggleDrawer());
```

File: src/screens/Drawer/DrawerContent.tsx [hunk 31]
```diff
@@ -0,0 +1,81 @@
...
+          routeName="Favorites"
+          setCurrentTab={setCurrentTab}
+          currentTab={currentTab}
+        />
+      </ButtonContainer>
+      <View>
+        <ButtonLogout onPress={handleGoogleSignOut}>
+          {loading ? (
+            <Loading
+              style={{transform: [{rotate: spin}]}}
```
Comment (IbraDajani) [inline src/screens/Drawer/DrawerContent.tsx:67]: Move this inline styling somewhere else like a custom hook for the animation

**Review by IbraDajani — CHANGES_REQUESTED** [review]

File: src/navigation/DrawerNavigationRoutes/index.tsx [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+import DrawerContent from '~/screens/Drawer/DrawerContent';
+import useDrawerNavigationController from './useDrawerNavigationController';
+
+const {Navigator, Screen} = createDrawerNavigator();
+
+const DrawerRoutes = () => {
+  const {screenOptions} = useDrawerNavigationController();
+  return (
+    <Navigator
+      drawerContent={() => <DrawerContent />}
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/navigation/DrawerNavigationRoutes/index.tsx]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/navigation/DrawerNavigationRoutes/index.tsx [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+      screenOptions={screenOptions}>
+      <Screen name="Home" component={Home} />
+      <Screen
+        options={{swipeEnabled: false}}
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/navigation/DrawerNavigationRoutes/index.tsx]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/DrawerContent.tsx (hunk 31 above)
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/DrawerContent.tsx]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/styles.ts [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+  align-items: center;
+  background-color: #f1f1f1;
+  justify-content: space-between;
+  overflow: hidden;
+  padding-right: 20px;
+`;
+export const ProfileContainer = styled.View`
+  align-items: center;
+`;
+export const AvatarProfile = styled(Avatar)``;
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/styles.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/styles.ts [hunk 35]
```diff
@@ -0,0 +1,47 @@
...
+export const ButtonContainer = styled.View`
+  flex-direction: column;
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/styles.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/styles.ts [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+  margin-bottom: 170px;
+`;
+export const BackgroundImage = styled.ImageBackground`
+  width: 285px;
+  height: 285px;
+  left: -175px;
+  top: -75px;
+  position: absolute;
+  opacity: 0.1;
+`;
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/styles.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/styles.ts [hunk 37]
```diff
@@ -0,0 +1,47 @@
...
+  overflow: hidden;
+  padding-right: 20px;
+`;
+export const ProfileContainer = styled.View`
+  align-items: center;
+`;
+export const AvatarProfile = styled(Avatar)``;
+export const ButtonContainer = styled.View`
+  flex-direction: column;
+  margin-bottom: 170px;
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/styles.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Drawer/useDrawerContentController.ts [hunk 38]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  Animated.loop(
+    Animated.timing(spinValue, {
+      toValue: 1,
+      duration: 1000,
+      easing: Easing.linear,
+      useNativeDriver: true,
+    }),
+  ).start();
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Drawer/useDrawerContentController.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Favorites/useFavoritesController.ts [hunk 39]
```diff
@@ -0,0 +1,11 @@
+import {DrawerActions, useNavigation} from '@react-navigation/native';
+
+const useFavoritesController = () => {
+  const navigation = useNavigation();
```
[the remark this answers is not in the document]
Comment (Leeosteves) [reply src/screens/Favorites/useFavoritesController.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Favorites/useFavoritesController.ts (hunk 24 above)
Comment (Leeosteves) [reply src/screens/Favorites/useFavoritesController.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Favorites/index.tsx (hunk 27 above)
Comment (Leeosteves) [reply src/screens/Favorites/index.tsx:13]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Favorites/styles.ts (hunk 28 above)
Comment (Leeosteves) [reply src/screens/Favorites/styles.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Home/useHomeController.ts (hunk 29 above)
Comment (Leeosteves) [reply src/screens/Home/useHomeController.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

File: src/screens/Home/useHomeController.ts (hunk 30 above)
Comment (Leeosteves) [reply src/screens/Home/useHomeController.ts]: resolved in commit 2c631c64d3008d3793416ddfe5451f0001937690

**Review by IbraDajani — APPROVED** [review]
🔥

**Review by IagoWag — APPROVED** [review]

**Pull request merged**